### DISCUSSION TITLE
Preview updated tasklist UI

### DISCUSSION THREAD

**Discussion by jaap-karssenberg** [zim-desktop-wiki/zim-desktop-wiki#1637] [Development]
For the next release I'm working on updating the tasklist UI. Work is still in a private branch, will push to develop when ready for testing. But here is a quick Preview.

Btw. the same branch has work on various tasklist related tickets - most notably #486 and #780 - so not strictly UI only.

![Screenshot from 2021-09-28 21-08-32](https://user-images.githubusercontent.com/1248286/135150945-a2d930f2-14c1-4a56-bb2d-5ae56f013a25.png)
Showing side pane with "active" list - same as what is now in the "flat list" context menu option

![Screenshot from 2021-09-28 21-08-36](https://user-images.githubusercontent.com/1248286/135151074-48db45d3-98ea-4675-a91c-f16aeb87e213.png)
Showing all selection options, "lists" are mutually exclusive (and e.g. have flat versus nested views), the other labels can be combined to create sub-selections.

The "GTD" inspired lists can be turned on in the plugin preferences, so default pane has less "lists".

![Screenshot from 2021-09-28 21-09-04](https://user-images.githubusercontent.com/1248286/135151239-f00880cb-2ee8-4dad-8be9-8e65eb5594a9.png)
Dialog - now actually a window - has same selection pane. It can be opened directly by the "pop out" icon in the pane.

![Screenshot from 2021-09-28 21-09-27](https://user-images.githubusercontent.com/1248286/135151331-91008707-5191-417f-9e76-fb7b10b595ae.png)
Showing sub-selection in dialog

**Comment by bruvellu** [comment]
Hi Jaap, nice updates, thanks for the preview. One aspect about the UI of the task list I just remembered is the current highlight colors (high=red, medium=orange, alert=yellow). They don't work very well with a dark theme:

![Screenshot from 2021-09-28 23-02-57](https://user-images.githubusercontent.com/361179/135166295-ea2ea5ce-ac4c-4e0f-bfa9-331b282fa3d8.png)

I don't think they need to be [configurable](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/master/zim/plugins/tasklist/gui.py#L350); a hard coded dark variant would already solve the contrast issue. Although it'll probably need a configuration toggle in the plugin preferences (unless the theme can be detected automatically). Not sure what's the best solution...

Cheers,
Bruno
